src/OrchardCore.Modules/OrchardCore.Search.Lucene/Handler/LuceneIndexingContentHandler.cs

```diff
@@ -88,31 +85,78 @@ private static async Task IndexingAsync(ShellScope scope, IEnumerable<ContentCon
 
                     if (indexSettings.IndexedContentTypes.Contains(context.ContentItem.ContentType) && !ignoreIndexedCulture)
                     {
-                        if (!indexSettings.IndexLatest && !publishedLoaded)
+                        if (context is RemoveContentContext)
                         {
-                            publishedLoaded = true;
-                            published = await contentManager.GetAsync(context.ContentItem.ContentItemId, VersionOptions.Published);
+                            await luceneIndexManager.DeleteDocumentsAsync(indexSettings.IndexName, new string[] { context.ContentItem.ContentItemId });
+                            continue;
                         }
 
-                        if (indexSettings.IndexLatest && !latestLoaded)
+                        async Task storeLuceneDocument(ContentItem contentItem)
```

Comment:
so many changes here. It's hard to follow. can you more this local function outside to reduce the noise?